server/src/immich/api-v1/asset/asset.service.ts

```diff
@@ -208,32 +211,20 @@ export class AssetService {
   }
 
   public async getAllAssets(authUser: AuthUserDto, dto: AssetSearchDto): Promise<AssetResponseDto[]> {
-    if (dto.userId && dto.userId !== authUser.id) {
-      await this.checkUserAccess(authUser, dto.userId);
-    }
+    await this.access.requirePermission(authUser, Permission.LIBRARY_READ, dto.userId || authUser.id);
```

Comment:
I'd prefer to extract variable `const userId = dto.userId || authUser.id` like in `getAssetByTimeBucket` and `getAssetCountByTimeBucket`